.github/workflows/checks.yml

```diff
@@ -0,0 +1,50 @@
+name: Checks
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  test_and_lint:
+    runs-on: macos-latest
```

Comment:
I believe macos version in gh will change very soon, to run macos-12. Just fyi (because some things may change)

Reply:
Changed it to include explicit version